### ISSUE TITLE
ByteString concat api

### ISSUE THREAD

**Issue by gengjiawen** [square/okio#295]
Hi, I want to use some feature like this, but i can't find similar api in ByteString. But I find it in https://android.googlesource.com/platform/external/okhttp/+/3c938a3/okio/src/main/java/okio/ByteString.java (Line 179). Do i miss something ?

**Comment by swankjesse** [comment]
```
ByteString a = ...;
ByteString b = ...;
ByteString ab = new Buffer().write(a).write(b).readByteString();
```

**Comment by gengjiawen** [comment]
Thanks for this, but I think concat api will be more intuitive and convenient.

**Comment by swankjesse** [comment]
A couple months ago I wrote a thing about what I called seductive code:
https://publicobject.com/2016/12/19/seductive-code/

I think this concat thing would _look_ convenient, but the calling code is actually worse. In particular I hate the idea of having developers choose between these alternatives:
```
ByteString ab = new Buffer().write(a).write(b).readByteString();
```

```
ByteString ab = a.concat(b);
```

Because although the 2nd one is definitely “more intuitive and convenient” it is also going to seduce you into doing this:

```
ByteString abc = a.concat(b).concat(c);
```
And that is definitely not good, not good at all. That’s really inefficient, and I want the normal way to be the efficient way.
